### PR TITLE
feat(schema): Update schema to include all from stoplight doc

### PR DESCRIPTION
**Pull request by chuckcarpenter**
This branch adds the schema information for all models that were documented in [Stoplight](https://stoplight.io/p/studio/gh/resideo/lantern). 

**What kind of change does this PR introduce?**

Refactors the schema file to include lots more things, changes naming, and removes a couple items that were either redundant or not used.

**Does this PR introduce a breaking change?**

No, since no one is using this file yet.

To see how it would look, I tested at https://apis.guru/graphql-voyager/
image

### CODE REVIEW

File: src/services/Temporary/schema.graphql [hunk 1]
```diff
@@ -2,86 +2,456 @@ scalar Date
...
+type NotificationChannels {
+  Email: Boolean
+  Push: Boolean
+  InAppPrompt: Boolean
+  SMS: Boolean
+}
+
+type UserDiscountGroup {
+  userID: String
+  userPriceGroup: String
```
[the remark this answers is not in the document]
Comment (WY-Resideo) [reply src/services/Temporary/schema.graphql]: I see this is outdated but to ensure we are tied out ... FYI my background is ERP / SQL / relational dbs, so I may be overworking something that just isn't needed in a graphQL context:

Assumptions
- user (master entity)
- subscriptionPricing (entity)

user and subscriptionPricing entities would both have userDiscountGroup as a field/drilldown.  Multiple userDiscountGroups could be defined.

Potential queries for userDiscountGroup:
- Give me a list of users that have userDiscountGroup (unique ID)
- Give me a list of (active) subscriptionPricing records with userDiscountGroup (unique ID)

My intent was to establish "Price" as one explicit concept (e.g. List price for X service is $4.99), and discounts as a separate/additional layer.

File: src/services/Temporary/schema.graphql [hunk 2]
```diff
@@ -2,86 +2,456 @@ scalar Date
...
+  firstName: String!
+  lastName: String!
+  address: Address!
+  contactEmail: String!
+  contactPhoneNumber: PhoneNumber
+  createdAt: DateTime
   dateOfBirth: Date
-  address: Location
-  phone: PhoneNumber
+  airPreference: Int
```
[the remark this answers is not in the document]
Comment (WY-Resideo) [reply src/services/Temporary/schema.graphql]: https://resideotechnologies.slack.com/archives/GMZ59J45N/p1568308439109900
Slack thread link included, but including comments here to ensure github has source of truth:

That is a true MarComm (or marketing-marketing) thing.  Preference or Index fields like that as an integer are intended to be profile info about the users interest level in a particular thing.  airInterest could be a better name.  In MarComm, they tend to use "Air Index" in their own CRM systems, but we can use a name that makes sense for the contract and display the expected term in the UI

Ultimately, it's a field we would track to know if the User likes or does not like conversations about Air Quality.  If they have a higher preference, we'd be more likely to send "lower priority" messages.  If they have a lower preference, we'd be more likely to only send "high priority" messages...

File: src/services/Temporary/schema.graphql (hunk 1 above)
[the remark this answers is not in the document]
Comment (chuckcarpenter) [reply src/services/Temporary/schema.graphql]: @WY-Resideo I believe you can get both of those queries by the users key on `UserDiscountGroup` and the `UserPricing` record it's set to. These are here: https://github.com/resideo/zeus-api/pull/27/files#diff-efadb8134463ae57399d78d3772feacbR110

[the remark this answers is not in the document]
Comment (chuckcarpenter) [reply src/services/Temporary/schema.graphql]: Does that make sense to you @shanewilson ?

File: src/services/Temporary/schema.graphql (hunk 2 above)
[the remark this answers is not in the document]
Comment (taras) [reply src/services/Temporary/schema.graphql]: @WY-Resideo do you know what is the range of those values? 

Although it does sounds like an Int is appropriate here.

File: src/services/Temporary/schema.graphql (hunk 1 above)
[the remark this answers is not in the document]
Comment (taras) [reply src/services/Temporary/schema.graphql]: https://github.com/resideo/zeus-api/pull/27#discussion_r323947853 this looks good to me

**Comment by WY-Resideo** [comment]
I don't know what Resideo uses (or plans to use) but 0-10 or 0-100 is common.  0-1000 is possible, I suppose number/decimal may be more appropriate than 'int'

In one of the versions of stoplight I was using, I recall I could only find 'int'
________________________________

**Comment by chuckcarpenter** [comment]
@shanewilson @taras are there any blockers to getting this first pass merged? Happy to change anything, but not seeing any actionable comments.

**Comment by taras** [comment]
I think we’re good to merge

**Pull request merged**